core/src/main/java/org/mule/runtime/core/internal/routing/UntilSuccessfulRouter.java

```diff
@@ -185,8 +189,8 @@
       } else { // Retries exhausted
         // Current context already pooped. No need to re-insert it
         LOGGER.error("Retry attempts exhausted. Failing...");
-        Throwable resolvedError = getThrowableFunction(ctx.event).apply(error);
         // Delete current context from event
+        Throwable resolvedError = getThrowableFunction(ctx.event).apply(error);
```

Comment:
is there any reason to move this line down?

Reply:
No reason...